### ISSUE TITLE
Automatically generate texture atlases from source objects

### ISSUE THREAD

**Issue by Frooxius** [Frooxius/NeosPublic#237]
Adding a feature that would take arbitrary input object and automatically atlas all possible materials would significantly simplify optimization of avatars, worlds and other arbitrary objects.

The strongly typed material definitions can be exploited to automatically extract information and even combine materials that use some different parameters (e.g. tint color) by premultiplying those into the texture.

This could enable workflows where user first customizes their avatar with multi-material setup and once finished run the automatic atlasing for optimization. This technique could also be require on mobile headsets.

**Comment by ProbablePrime** [comment]
I was about to suggest this, thank you :)

**Comment by Frooxius** [comment]
@ProbablePrime Yeah I saw the Nanachi setup and I just thought it would be ideal for that! Wanted to add this for a while, but this is a good excuse to log it :)

**Comment by eduprimowl** [comment]
Please add soon
